app/ide-desktop/lib/client/src/index.ts

```diff
@@ -50,7 +51,7 @@ class App {
             this.setChromeOptions(chromeOptions)
             security.enableAll()
             electron.app.on('before-quit', () => (this.isQuitting = true))
-            electron.app.whenReady().then(() => this.main(windowSize))
+            electron.app.on('ready', () => this.main(windowSize))
```

Comment:
IIRC without this change `open-url` events didn't fire when deep links were clicked. I'm still not 100% sure _why_ that happened. I'll investigate the issue and try to resolve it without this change, if possible. I'll document why this change was made, if it isn't.

Reply:
Yeah so without this change the application freezes on startup. The spinner gets and stays stuck. I'm unable to diagnose why since I'm not able to get `showLogs()` to work once this happens. Any advice on how to investigate the cause of such a crash? @wdanilo

Reply:
Super strange. Let's do this:

1. Add this code:

```
electron.app.whenReady().then(() => console.log("whenReady!"))
electron.app.on('ready', () => console.log('ready'))
```

2. Leave the version with `whenReady` and try replicating the app stuck on loader screen.
3. Tell me what you see in terminal.
4. Until its resolved, lets change it to `on('ready', ...)` + lets create issue for it on our issue tracker and link it in the codebase.
5. When we will have that confirmed / diagnosed, we will report that in electron's GitHub.

Reply:
Sounds good. Opened as issue #5851 and added a comment to the codebase.